pandas/core/frame.py

```diff
@@ -564,41 +564,59 @@ def __init__(
         if isinstance(data, DataFrame):
             data = data._mgr
 
-        if isinstance(data, (BlockManager, ArrayManager)):
-            if index is None and columns is None and dtype is None and copy is False:
-                # GH#33357 fastpath
-                NDFrame.__init__(self, data)
-                return
+        # first check if a Manager is passed without any other arguments
+        # -> use fastpath (without checking Manager type)
+        if (
+            index is None
+            and columns is None
+            and dtype is None
+            and copy is False
+            and isinstance(data, (BlockManager, ArrayManager))
+        ):
+            # GH#33357 fastpath
+            NDFrame.__init__(self, data)
+            return
 
+        manager = get_option("mode.data_manager")
+
+        if isinstance(data, (BlockManager, ArrayManager)):
             mgr = self._init_mgr(
                 data, axes={"index": index, "columns": columns}, dtype=dtype, copy=copy
             )
 
         elif isinstance(data, dict):
-            mgr = init_dict(data, index, columns, dtype=dtype)
+            mgr = init_dict(data, index, columns, dtype=dtype, typ=manager)
```

Comment:
can you rename typ -> manager?

Reply:
Note this is the same keyword as I use for `arrays_to_mgr`, `mgr_to_mgr`, etc, where I think the `typ` keyword makes sense (it's already clear from the name of the function that it is creating a manager).

Alternatively, we could make the `init_dict` function name more explicit (and consistent with the others) and eg rename to `dict_to_mgr`. I am happy to do this in a separate PR.

Reply:
ok yeah +1 on the rename (followon ok ) and anything to make the keyword more obvious

Reply:
-> https://github.com/pandas-dev/pandas/pull/40074